### PR TITLE
Fix buffer attributes build on windows

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: src/core/buffer_attributes.h [hunk 1]
```diff
@@ -24,6 +24,7 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+#include <iterator>
```
Comment (tanmayv25) [inline src/core/buffer_attributes.h:27]: Why is this needed?

Comment (Tabrizian) [reply src/core/buffer_attributes.h:27]: For the `https://www.cplusplus.com/reference/iterator/back_inserter/` used in the buffer_attributes.cc.

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: src/servers/shared_memory_manager.cc [hunk 2]
```diff
@@ -53,6 +53,16 @@ SharedMemoryManager::RegisterCUDASharedMemory(
       std::string("Shared memory feature is currently not supported on Windows")
           .c_str());
 }
+
+TRITONSERVER_Error*
+SharedMemoryManager::GetCUDAHandle(
+    const std::string& name, cudaIpcMemHandle_t** cuda_mem_handle)
+{
+  return TRITONSERVER_ErrorNew(
```
[the remark this answers is not in the document]
Comment (GuanLuo) [reply src/servers/shared_memory_manager.cc:61]: Macro may not save us much here, you still need to pass in the error code and message for the use of macro.

[the remark this answers is not in the document]
Comment (Tabrizian) [reply src/servers/shared_memory_manager.cc:61]: Yeah, I think using macros we might still have a lot of repetition.

**Review by deadeyegoodwin — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Review by CoderHam — APPROVED** [review]
